vignettes/SEIRD_Model_Compartments.Rmd

```diff
@@ -0,0 +1,199 @@
+$$
+F = \left(\begin{array}{cc} 
+0 & S_0\beta_{I_a} & S_0\beta_{I_m} & S_0\beta_{I_s}\\
+0 & 0 & 0 & 0\\
+0 & 0 & 0 & 0\\
+0 & 0 & 0 & 0
+\end{array}\right), 
+V = \left(\begin{array}{cc} 
+\kappa & 0 & 0 & 0\\
+-\kappa \eta_{E->I_a} & \gamma_{I_a} + \mu_{I_a} & 0 & 0\\
+-\kappa \eta_{E->I_m} & 0 & \gamma_{I_m} + \mu_{I_m} & 0\\
+-\kappa \eta_{E->I_s} & 0 & 0 & \gamma_{I_s} + \mu_{I_s}
+\end{array}\right)
+$$
+where $S_0$ is the initial number of susceptibles, as in assumption the initial conditions $x_0 = \{S_0, E_0, ...\}$ can be seen as a DFE given that $S_0, E_0>0$ and $E_0 \ll S_0$, and $x_{0i}=0$ for all other compartments except $S_0$ and $E_0$ . $\mathscr{R_0}$ can then be calculated using the equation $\mathscr{R_0} = \rho(FV^{-1})$.
+
+## The `SEIaImIsRD` class
+In this section we will walk through an example of how to use the `SEIaImIsRD`
+class.
+
+To create a new `SEIaImIsRD` object, run:
+```{r}
+model <- SEIaImIsRD()
+```
+
+Next, we assign values for the fractions of the population.
+The initial conditions must sum to one.
+```{r}
+S <- 0.99
+E <- 0.01
+I_asymptomatic <- 0
+I_mild <- 0
+I_severe <- 0
+R <- 0
+D_cumulative <- 0
+```
+
+Then, we define the transition rates:
+```{r}
+beta <- list(i_asymptomatic = 0.80, i_mild = 0.90, i_severe = 1.00)
+kappa <- 0.50
+omega <- 0.01
+p_symptom <- list(i_mild = 0.35, i_severe = 0.05)
+gamma <- list(i_asymptomatic = 0.9, i_mild = 0.50, i_severe = 0.05)
+mu <- list(i_asymptomatic = 0.005, i_mild = 0.05, i_severe = 0.30)
+```
+
+And, we set time for the simulation:
+```{r}
+t <- seq(0, 2000, by = 1)
+```
+
+We initialise our model with the parameters defined above:
+```{r}
+model <- set_init(object = model, S = S, E = E, I_asymptomatic = I_asymptomatic, I_mild = I_mild,
+                    I_severe = I_severe, R = R, D = D_cumulative,
+                    beta = beta, kappa = kappa, omega = omega,
+                    p_symptom = p_symptom, gamma = gamma, mu = mu)
+```
+
+We use the ode solver, plot the relaxation into the steady state and calculate the R0 of the system by calling the function `R0_SEIaImIsRD`:
+```{r}
+model <- ode_simulate(model, t)
+model <- R0_SEIaImIsRD(model)
+p <- plot_dataframe(model@output, x = "time", y = "fraction", c = "population_group")
+print(p)
+print(paste0("model: R0=", model@R0))
+```
+
+Comparison with no natural loss of immunity (no transmission from R to S; omega=0)
+```{r}
+model1 <- set_init(object = model, S = S, E = E, I_asymptomatic = I_asymptomatic, I_mild = I_mild,
+                    I_severe = I_severe, R = R, D = D_cumulative,
+                    beta = beta, kappa = kappa, omega = 0.00,
+                    p_symptom = p_symptom, gamma = gamma, mu = mu)
+
+model1 <- ode_simulate(model1, t)
+model1 <- R0_SEIaImIsRD(model1)
+p1 <- plot_dataframe(model1@output, x = "time", y = "fraction", c = "population_group")
+print(p1)
+print(paste0("model1: R0=", model1@R0))
+```
+
+Comparison with altered beta values:
```

Comment:
Maybe add a short description on what beta is, like what you did for omega above.